janus-idp/backstage-showcase
# chore(auth): make oidc auth provider resolvers configurable

## Description

Adds resolver configurations for the oidc auth provider. Users can now specify the resolver the oidc auth provider uses. Currently supports the following:
- resolver: preferredUsernameMatchingUserEntityName
- resolver: emailMatchingUserEntityProfileEmail
- resolver: emailLocalPartMatchingUserEntityName

Using `preferredUsernameMatchingUserEntityName` should allow proper mapping to users ingested via the [keycloak-backend](https://github.com/janus-idp/backstage-plugins/tree/main/plugins/keycloak-backend) plugin with no additional configurations.

## Which issue(s) does this PR fix

- Fixes [RHIDP-2272](https://issues.redhat.com/browse/RHIDP-2272)

## PR acceptance criteria

Please make sure that the following steps are complete:


## How to test changes / Special notes to the reviewer
- Setup a `keycloak` instance and create a user with a username (ex: test-user) and an email (ex: tester@mail.com)
- Setup the `keycloak-backend` plugin to ingest users/groups
- Configure OIDC auth provider to connect to the keycloak instance
- Add the `preferredUsernameMatchingUserEntityName` resolver
- Change `signInPage` to `oidc`
- Sign-in with OIDC in the frontend and you should be able to login and be associated with the keycloak user ingested via the keycloak-backend.
